pareto/tests/test_strategic_model.py

```diff
@@ -440,6 +448,131 @@ def _call_model_with_config(config_dict):
+
+    assert isinstance(mh, pyo.ConcreteModel)
+    assert isinstance(mh.hydraulics, pyo.Block)
+    assert isinstance(mh.hydraulics.v_Pressure, pyo.Var)
+    assert isinstance(mh.hydraulics.v_PumpHead, pyo.Var)
+    assert isinstance(mh.hydraulics.vb_Y_Pump, pyo.Var)
+    assert isinstance(mh.hydraulics.v_ValveHead, pyo.Var)
+    assert isinstance(mh.hydraulics.v_PumpCost, pyo.Var)
+    assert isinstance(mh.hydraulics.p_HW_material_factor_pipeline, pyo.Param)
+    assert isinstance(mh.hydraulics.p_rhog, pyo.Param)
+    assert isinstance(mh.hydraulics.p_PumpFixedCost, pyo.Param)
+    assert isinstance(mh.hydraulics.p_ElectricityCost, pyo.Param)
+    assert isinstance(mh.hydraulics.p_PumpEfficiency, pyo.Param)
+    assert isinstance(mh.hydraulics.p_MotorEfficiency, pyo.Param)
+    assert isinstance(mh.hydraulics.p_WellPressure, pyo.Param)
+    assert isinstance(mh.hydraulics.p_effective_Pipeline_diameter, pyo.Param)
+    assert isinstance(mh.hydraulics.p_Min_AOP, pyo.Param)
+    assert isinstance(mh.hydraulics.p_Max_AOP, pyo.Param)
+    assert isinstance(mh.hydraulics.p_HW_loss, pyo.Param)
+    assert isinstance(mh.hydraulics.objective, pyo.Objective)
+    assert isinstance(mh.hydraulics.NodePressure, pyo.Constraint)
+    assert isinstance(mh.hydraulics.MAOPressure, pyo.Constraint)
+    assert isinstance(mh.hydraulics.PumpCostEq, pyo.Constraint)
+    assert isinstance(mh.hydraulics.HydraulicsCostEq, pyo.Constraint)
+    assert isinstance(mh.hydraulics.PumpHeadCons, pyo.Constraint)
+
+
+# if solver cbc exists @solver
+@pytest.mark.component
+def test_run_hydraulics_post_process_reduced_strategic_model(
+    build_reduced_strategic_model,
+):
+    m = build_reduced_strategic_model(
+        config_dict={
+            "objective": Objectives.cost,
+            "pipeline_cost": PipelineCost.distance_based,
+            "pipeline_capacity": PipelineCapacity.input,
+            "hydraulics": Hydraulics.post_process,
+            "node_capacity": True,
+            "water_quality": WaterQuality.false,
+            "removal_efficiency_method": RemovalEfficiencyMethod.concentration_based,
+        }
+    )
+
+    options = {
+        "deactivate_slacks": True,
+        "scale_model": False,
+        "scaling_factor": 1000,
+        "running_time": 60 * 5,
+        "gap": 0,
+    }
+    results = solve_model(model=m, options=options)
+
+    assert results.solver.termination_condition == pyo.TerminationCondition.optimal
+    assert results.solver.status == pyo.SolverStatus.ok
+    assert degrees_of_freedom(m) == 11789
+    # solutions obtained from running the reduced generic case study
+    assert pytest.approx(88199.598, abs=1e-1) == pyo.value(m.v_Z)
+    assert pytest.approx(26.067, abs=1e-1) == pyo.value(m.hydraulics.v_Z_HydrualicsCost)
+    assert pytest.approx(24, abs=1e-1) == pyo.value(
+        sum(m.hydraulics.vb_Y_Pump[key] for key in m.s_LLA)
+    )
+    # feasibility test is not working with Cbc so, disabling for now until a fix is found
```

Comment:
The infeasibility issue returned by CBC primarily stems from the violation of the max allowable pressure constraint (mh.MAOPressure). This, I believe, is predominantly due to the scaling challenge associated with the model's pressure unit being in Pascal, which inherently presents large numerical values.

To mitigate this, we can consider two potential solutions:

1) Change the pressure unit from Pascal to bar, adjust the initial value of rhog from 9.8 * 1000 to 9.8 * 1000 * 1e-5, and revise the PumpCostRule as follows:
def PumpCostRule(b, l1, l2):
    constraint = b.v_PumpCost[l1, l2] == mh.p_PumpFixedCost * mh.vb_Y_Pump[l1, l2] + ((mh.p_ElectricityCost / 3.6e6) * mh.p_rhog * 1e5 * sum(b.v_PumpHead[l1, l2, t] * pyunits.convert_value(value(model.v_F_Piped[l1, l2, t]), from_units=model.model_units["volume_time"], to_units=pyunits.meter**3 / pyunits.h) for t in model.s_T))
    return process_constraint(constraint)

2) Alternatively, we can multiply both sides of the mh.MAOPressure and nodePressureRule constraints by 1e-5 to effectively address the scaling issue.

I believe adopting either of these approaches should also contribute to resolving issue #268